src/overlays/actors/ovl_En_Peehat/z_en_peehat.c

```diff
@@ -96,98 +167,795 @@ static DamageTable D_80899510 = {
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Peehat/func_80898594.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Peehat/func_80898654.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Peehat/func_808986A4.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Peehat/func_8089874C.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Peehat/EnPeehat_Update.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Peehat/func_80898E74.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Peehat/func_80899024.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Peehat/EnPeehat_Draw.s")
+void EnPeehat_Init(Actor* thisx, GlobalContext* globalCtx) {
+    EnPeehat* this = THIS;
+
+    Actor_ProcessInitChain(&this->actor, sInitChain);
+    SkelAnime_Init(globalCtx, &this->skelAnime, &D_06001C80, &D_060009C4, this->jointTable, this->morphTable, 24);
+    ActorShape_Init(&this->actor.shape, 0.0f, func_800B3FC0, 27.0f);
+    this->unk_2B0 = 0;
+    Math_Vec3f_Copy(&this->actor.focus.pos, &this->actor.world.pos);
+    this->actor.floorHeight = this->actor.world.pos.y;
+    Collider_InitAndSetCylinder(globalCtx, &this->colliderCylinder, &this->actor, &sCylinderInit);
+    Collider_InitAndSetSphere(globalCtx, &this->colliderSphere, &this->actor, &sSphereInit);
+    Collider_InitAndSetTris(globalCtx, &this->colliderTris, &this->actor, &sTrisInit, this->colliderTriElements);
+
+    if (this->actor.params == 0) {
+        CollisionCheck_SetInfo2(&this->actor.colChkInfo, &sDamageTable, &sColChkInfoInit1);
+        if (gSaveContext.isNight) {
+            this->actor.shape.yOffset = -1000.0f;
+        }
+        Actor_SetScale(&this->actor, 0.036f);
+        this->actor.hintId = 0x48;
+        func_80897498(this);
+    } else {
+        CollisionCheck_SetInfo2(&this->actor.colChkInfo, &sDamageTable, &sColChkInfoInit2);
+        this->actor.scale.z = this->actor.scale.x = 0.006f;
+        this->actor.scale.y = 0.003f;
+        this->actor.velocity.y = 6.0f;
+
+        this->colliderCylinder.dim.radius = 20;
+        this->colliderCylinder.dim.height = 15;
+        this->colliderCylinder.dim.yShift = -5;
+        this->actor.hintId = 0x49;
+        this->colliderCylinder.base.ocFlags1 &= ~OC1_ON;
+
+        func_80897A34(this);
+    }
+}
+
+typedef struct {
+    Actor actor;
+    char unk_144[0x168];
+    u8 unk_2AC;
+} EnPeehatUnkParent;
+
+void EnPeehat_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    EnPeehat* this = THIS;
+
+    Collider_DestroyCylinder(globalCtx, &this->colliderCylinder);
+    Collider_DestroySphere(globalCtx, &this->colliderSphere);
+    Collider_DestroyTris(globalCtx, &this->colliderTris);
+    if (this->actor.params != 0) {
+        EnPeehatUnkParent* parent = (EnPeehatUnkParent*)this->actor.parent;
+
+        if ((parent != NULL) && (parent->actor.update != NULL)) {
+            parent->unk_2AC--;
+        }
+    }
+}
```

Comment:
```suggestion

void EnPeehat_Destroy(Actor* thisx, GlobalContext* globalCtx) {
    EnPeehat* this = THIS;

    Collider_DestroyCylinder(globalCtx, &this->colliderCylinder);
    Collider_DestroySphere(globalCtx, &this->colliderSphere);
    Collider_DestroyTris(globalCtx, &this->colliderTris);
    if (this->actor.params != 0) {
        EnPeehat* parent = (EnPeehat*)this->actor.parent;

        if ((parent != NULL) && (parent->actor.update != NULL)) {
            parent->unk_2AC--;
        }
    }
}
```
A peehat will spawn itself as child for the mini peehat's it spawns. So this `EnPeehatUnkParent` can be removed